src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const getYearFormat = (year) => {
+    if (year.length === 2) {
+      const shortYear = parseInt(year, 10);
+
+      return shortYear < 30 ? `20${year}` : `19${year}`;
+    }
+
+    return year;
+  };
+
+  const parseDate = (dateString, format) => {
+    const parts = dateString.split(format[format.length - 1]);
+    const dateObj = {};
+
+    format.forEach((part, index) => {
+      dateObj[part] = parts[index];
+    });
+
+    return dateObj;
+  };
+
+  const formatDatePart = (dateObj, format) => {
+    return format.map((part) => dateObj[part]).join(format[format.length - 1]);
```

Comment:
you can remove curly braces and return word:

`const formatDatePart = (dateObj, format) => format.map((part) => dateObj[part]).join(format[format.length - 1]);`